### PR TITLE
MaaS Template Refactor

### PR DESCRIPTION
**Pull request by sigmavirus24**

### CODE REVIEW

**Comment by stevelle** [comment]
see also https://github.com/rcbops/rpc-openstack/pull/530

**Comment by sigmavirus24** [comment]
In initial upgrade testing with this PR, @xeregin seems to have found that we're triggering duplicate entries.

File: rpcd/playbooks/maas-20151013-1-requirements.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      when: inventory_hostname == groups['hosts'][0]
+    - name: Locate evidence of prior MaaS migration
+      local_action: stat path={{ migrations_dir }}/maas-20151013
+      when: inventory_hostname == groups['hosts'][0]
+      register: maas_upgraded_result
+      changed_when: False
+    - name: Stop if migration has been run
+      fail: msg="MaaS has been previously run migration 20151013"
+      when:
+        - maas_upgraded_result is defined
```
Comment (sigmavirus24) [inline rpcd/playbooks/maas-20151013-1-requirements.yml]: This condition does not work. Needs to be

```yaml
when: >
  maas_upgraded_result is defined and maas_upgraded_result.stat.exists | bool
```

File: rpcd/playbooks/maas-20151013-1-requirements.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        - maas_upgraded_result.stat.exists | bool
+    - name: Locate existing Rackspace monitoring agent
+      stat: path=/etc/rackspace-monitoring-agent.cfg
+      register: raxmon
+      changed_when: False
+    - name: Halt if monitoring agent not found
+      fail: msg="MaaS is not installed and configured already"
+      when:
+       - raxmon is defined
```
Comment (sigmavirus24) [inline rpcd/playbooks/maas-20151013-1-requirements.yml]: Likewise here

**Comment by stevelle** [comment]
Good.  Defects found!
How do you feel about the plan of:  1) open an issue and 2) resolve on the feature branch and 3) re-merge from there?

**Comment by sigmavirus24** [comment]
That plan works for me. @stevelle we're just *still* trying to test out the upgrade work.

I have a question for you when you get back about how/when we restart the rackspace-monitoring-agent

**Comment by sigmavirus24** [comment]
This backport is unnecessary apparently. The kilo branch is presently intended to be used for 11.0.x releases and this is not intended for one of those releases.

**Pull request closed without merging**